src/textual_summary/textual_group.jsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import GenericGroup from './generic_group';
+import TagGroup from './tag_group';
+import SimpleTable from './simple_table';
+import OperationRanges from './operation_ranges';
+import MultilinkTable from './multilink_table';
+import TableListView from './table_list_view';
+import EmptyGroup from './empty_group';
+
+const renderComponent = (props) => {
```

Comment:
Maybe rewriting this to hash might be better than using switch
```jsx
const componentMapper = {
  GenericGroup: props => (
    <GenericGroup onClick={props.onClick} items={props.group.items} title={props.group.title} />
  ),
  TagGroup: props => <TagGroup items={props.group.items} title={props.group.title} />,
  SimpleTable: props => (
    <SimpleTable
      labels={props.group.labels}
      rows={props.group.rows}
      title={props.group.title}
    />
  ),
  OperationRanges: props => <OperationRanges items={props.group.items} title={props.group.title} />,
  MultilinkTable: props => (
    <MultilinkTable
      onClick={props.onClick}
      items={props.group.items}
      title={props.group.title}
    />
  ),
  TableListView: props => (
    <TableListView
      rowLabel={props.group.rowLabel}
      onClick={props.onClick}
      title={props.group.title}
      headers={props.group.headers}
      values={props.group.values}
      colOrder={props.group.colOrder}
    />
  ),
  EmptyGroup: props => <EmptyGroup title={props.group.title} text={props.group.text} />
};
```
That way you can specify which component is using which propTypes
```javascript
componentMapper.GenericGroup.propTypes = {
  group: PropTypes.shape({
    title: PropTypes.string.isRequired,
    items: PropTypes.any,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
};

componentMapper.TagGroup.propTypes = {
  group: PropTypes.shape({
    title: PropTypes.string.isRequired,
    items: PropTypes.any,
  }).isRequired,
};
```
Or use [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) and inherit these properties.

Reply:
> Maybe rewriting this to hash might be better than using switch

If you mean a simple hash, that it would not work, because a hash is not lazy evaluated. All the branches would get called and then the apropriate result would be selected. So that approach would crash on the propTypes mismatch (different case branches need diffferent properties).

Reply:
Oh, you mean a hash of functions. Then yes, it woudl work but is a hash of functions actually better than what I have now?

Reply:
Anyway. If it's a 'maybe changing it will look better' then I am not changing it unless you insist. I see the difference as marginal.